New theme submit page

itsMapleLeaf opened the pull request:
![image](https://user-images.githubusercontent.com/19603573/198696066-3db96083-7f26-4487-861b-0021b9e9e7d6.png)

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+            </Field>
+        </div>
+        <InputField
+            label="Live demo URL"
```
`type="url"` should give us basic browser data validation for the URLs

itsMapleLeaf replied:
Didn't know that was a thing! :+1:

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+            placeholder="https://example.com/theme-demo"
+        />
+        <InputField
+            label="Public repo URL (required for free themes)"
```
Same here for `type="url"`

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+        document.querySelector<HTMLInputElement>('[data-free-switch]')
+    const repoUrl = document.querySelector<HTMLInputElement>('[data-repo-url]')
+
+    const updateRepoUrlRequired = () => {
```
Oh, nice!  Let's remove the "(required for free themes)" text too and toggle a red * here when it's required

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+        </ul>
+    </section>
+
+    <form
```
Lets add their honeypot spam filtering too, it's not perfect but should block the dumb bots out there

https://docs.netlify.com/forms/spam-filters/#honeypot-field

itsMapleLeaf replied:
That's a really neat technique actually

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+    <form
+        method="post"
+        class="max-w-screen-sm flex flex-col items-center gap-4 mx-auto py-8 px-4"
+        data-netlify="true"
```
Lets add a custom success page that points to `/themes/submit/success`

https://docs.netlify.com/forms/setup/#success-messages

itsMapleLeaf replied:
Ah yeah, forgot about that too

tony-sull commented on src/pages/themes/submit.astro:
```diff
@@ -0,0 +1,105 @@
+    </PageTitle>
+
+    <section class="max-w-screen-md mx-auto px-4">
+        <h3 class="text-3xl mt-8 mb-4">Best practices</h3>
```
This should be an `<h2>` since there isn't another one on the page

tony-sull reviewed (changes requested):
It's looking great! Really like the image upload component, nice to see there was a clean no-JS fallback there

Added a few comments and extra features we should add before shipping

A few design ideas I was testing out below, not a great way to include them in a review but I can push them up if we want to keep them :+1:

<img width="1272" alt="Screen Shot 2022-10-28 at 2 32 23 PM" src="https://user-images.githubusercontent.com/15836226/198717715-d435b690-1c9d-4564-93ba-414961851466.png">

itsMapleLeaf commented:
For the image input, I converted it to a Preact component, because the logic for it was getting tangled and hard to work with :sweat_smile: I also updated it so the "Please choose a file" bubble actually shows if you try to submit without one, which doesn't show when it has the `hidden` attribute

tony-sull reviewed (approved):
looks good!  Crazy that Netlify Functions doesn't support mutli-file uploads
